runtime/lua/vim/iter.lua

```diff
@@ -179,6 +208,60 @@ function ListIter.filter(self, f)
   return self
 end
 
+--- Flattens a |list-iterator| pipeline, un-nesting nested values.
+--- Errors if it attempts to flatten a dict-like value
+---
+--- Example:
+---
+--- ```lua
+--- vim.iter({1, {2}, {{3}}}):flatten():totable()
+--- ```
+--- (returns `{1, 2, {3}}`)
+---
+--- ```lua
+--- vim.iter({1, {{a = 2}}, {3}}):flatten():totable()
+--- ```
+--- (returns `{1, {a = 2}, 3}`)
+---
+--- ```lua
+--- vim.iter({1, {{a = 2}}, {3}}):flatten(2):totable()
+--- ```
+--- errors because it attempts to flatten a dict-like table ({a = 2})
+---@param depth number|nil Depth to which |list-iterator| pipeline should be flattened
```

Comment:
```suggestion
---@param depth? number Depth to which |list-iterator| should be flattened
```

Reply:
I think it should be `@param depth number?`, per https://luals.github.io/wiki/annotations/#documenting-types:

>Adding a question mark (?) after a type like boolean? or number? is the same as saying boolean|nil or number|nil.

Reply:
In general, if there's no observable difference in tooling (both luals and our doc generator script), either variant is acceptable, but the usage should be consistent at least within a file.